firmware/baseband/proc_btlerx.cpp

```diff
@@ -26,245 +27,476 @@
+
+//     //AA = reorder_bytes_str( payload_bytes((2*6+2*6+1):(2*6+2*6+2*4)) );
+//     payload_type_5->AA[0] = payload_byte[15];
+//     payload_type_5->AA[1] = payload_byte[14];
+//     payload_type_5->AA[2] = payload_byte[13];
+//     payload_type_5->AA[3] = payload_byte[12];
+
+//     //CRCInit = payload_bytes((2*6+2*6+2*4+1):(2*6+2*6+2*4+2*3));
+//     payload_type_5->CRCInit = ( payload_byte[16] );
+//     payload_type_5->CRCInit = ( (payload_type_5->CRCInit << 8) | payload_byte[17] );
+//     payload_type_5->CRCInit = ( (payload_type_5->CRCInit << 8) | payload_byte[18] );
+
+//     //WinSize = payload_bytes((2*6+2*6+2*4+2*3+1):(2*6+2*6+2*4+2*3+2*1));
+//     payload_type_5->WinSize = payload_byte[19];
+
+//     //WinOffset = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+1):(2*6+2*6+2*4+2*3+2*1+2*2)) );
+//     payload_type_5->WinOffset = ( payload_byte[21] );
+//     payload_type_5->WinOffset = ( (payload_type_5->WinOffset << 8) | payload_byte[20] );
+
+//     //Interval = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+2*2+1):(2*6+2*6+2*4+2*3+2*1+2*2+2*2)) );
+//     payload_type_5->Interval = ( payload_byte[23] );
+//     payload_type_5->Interval = ( (payload_type_5->Interval << 8) | payload_byte[22] );
+
+//     //Latency = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+2*2+2*2+1):(2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2)) );
+//     payload_type_5->Latency = ( payload_byte[25] );
+//     payload_type_5->Latency = ( (payload_type_5->Latency << 8) | payload_byte[24] );
+
+//     //Timeout = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2+1):(2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2+2*2)) );
+//     payload_type_5->Timeout = ( payload_byte[27] );
+//     payload_type_5->Timeout = ( (payload_type_5->Timeout << 8) | payload_byte[26] );
+
+//     //ChM = reorder_bytes_str( payload_bytes((2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2+2*2+1):(2*6+2*6+2*4+2*3+2*1+2*2+2*2+2*2+2*2+2*5)) );
+//     payload_type_5->ChM[0] = payload_byte[32];
+//     payload_type_5->ChM[1] = payload_byte[31];
+//     payload_type_5->ChM[2] = payload_byte[30];
+//     payload_type_5->ChM[3] = payload_byte[29];
+//     payload_type_5->ChM[4] = payload_byte[28];
+
+//     //tmp_bits = payload_bits((end-7) : end);
+//     //Hop = num2str( bi2de(tmp_bits(1:5), 'right-msb') );
+//     //SCA = num2str( bi2de(tmp_bits(6:end), 'right-msb') );
+//     payload_type_5->Hop = (payload_byte[33]&0x1F);
+//     payload_type_5->SCA = ((payload_byte[33]>>5)&0x07);
+//    }
+//     else
+//     {
+//    //TODO: Handle Unknown PDU.
+//      payload_type_R = (ADV_PDU_PAYLOAD_TYPE_R *)adv_pdu_payload;
+//      memcpy(payload_type_R->payload_byte, payload_byte, num_payload_byte);
+//     return(-1);
+//    }
+
+    return 0;
+}
+
 void BTLERxProcessor::execute(const buffer_c8_t& buffer) {
     if (!configured) return;
 
-    // FM demodulation
+    // Pulled this implementation from channel_stats_collector.c to time slice a specific packet's dB.
```

Comment:
nit: consider pulling this apart into functions that help "name" chunks of work.
Just makes the code easier to read and helps the code describe itself.